examples/tensorflow/multi-gpu-tensorflow.ipynb

```diff
@@ -0,0 +1,207 @@
+    "\n",
+    "This dataset constitutes 40,000+ birds and has been taken from [kaggle](https://www.kaggle.com/gpiosenka/100-bird-species).\n",
+    "\n",
+    "We recommend to spin up a bigger Saturn Cloud instance now, with multiple GPUs, so we can distribute the training work across these and have them working simultaneously, all training the same model.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "440e847b-0887-4159-8a06-6fcaef016c70",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tensorflow as tf\n",
+    "import keras\n",
+    "import time\n",
+    "import matplotlib.pyplot as plt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7a87f3bb-3b2f-497f-bdb5-fe1977949030",
+   "metadata": {},
+   "source": [
+    "The dataset originally had 285 classes. We have taken subset of this data which has 61 classes. The data is stored in AWS S3. The first time you run this job, you'll need to download the training and test data which will be saved at path `dataset/birds/`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5679b4bb-2f79-4327-bb18-d447c9fc4397",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import s3fs\n",
+    "\n",
+    "s3 = s3fs.S3FileSystem(anon=True)\n",
+    "_ = s3.get(\n",
+    "    rpath=\"s3://saturn-public-data/100-bird-species/100-bird-species/*/*/*.jpg\",\n",
+    "    lpath=\"dataset/birds/\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1ffc316c-afca-498c-a1e0-a6859a7fbe59",
+   "metadata": {},
+   "source": [
+    "Our dataset has already neatly separated out training, test, and validation samples. In code below we are constructing Keras data object for training and validation set using `keras.preprocessing.image_dataset_from_directory`. We have chosen Adam optimizer and have set learning rate to 0.02.  We are training our classifier with ResNet50 architecture, which has 48 Convolution layers along with 1 MaxPool and 1 Average Pool layer. The model is compiled, trained and saved at path `model/keras_single/`. This is actually incredibly similar to our single GPU approach. We’re just applying the Mirrored Strategy scope around our model definition and compilation stage, so that TensorFlow knows this model is to be trained on multiple GPUs."
```

Comment:
```suggestion
    "Our dataset has already neatly separated training, test, and validation samples. In the code below, we construct Keras data objects for training and validation sets using `keras.preprocessing.image_dataset_from_directory`. We chose the Adam optimizer and set the learning rate to 0.02. We train our classifier with ResNet50 model architecture, which has 48 Convolution layers along with 1 MaxPool and 1 Average Pool layer. The model is compiled, trained, and saved in `model/keras_single/`. This function is actually incredibly similar than our single GPU approach. We’re simply applying the Mirrored Strategy scope around our model definition and compilation stage, so that TensorFlow knows this model is to be trained on multiple GPUs."
```